### PR TITLE
lint: Integrate golangci-lint

### LINKED ISSUE #515: Integrate golang-lint
**What happened**:
We do small linting, would be good to integrate the one from golang-ci itself, it's used at controller-runtime project

https://github.com/kubernetes-sigs/controller-runtime/blob/45db7e68271f72d1f76b2222648bb7ade5ef8f2e/Makefile#L75

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- `NodeNetworkState` on affected nodes (use `kubectl get nodenetworkstate <node_name> -o yaml`):
- Problematic `NodeNetworkConfigurationPolicy`:
- kubernetes-nmstate image (use `kubectl get pods --all-namespaces -l app=kubernetes-nmstate -o jsonpath='{.items[0].spec.containers[0].image}'`):
- NetworkManager version (use `nmcli --version`)
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Others:

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind enhancement

**What this PR does / why we need it**:
There is plenty of linters out there for golang, the golangci-lint allow
to select wich ones and how you want to run.

**Special notes for your reviewer**:

Closes #515 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -115,8 +115,8 @@ whitespace-check:
 gofmt-check: $(GO)
 	test -z "`$(GOFMT) -l cmd/ test/ hack/ api/ controllers/ pkg/`" || ($(GOFMT) -l cmd/ test/ hack/ api/ controllers/ pkg/ && exit 1)
 
-$(GO):
-	hack/install-go.sh $(BIN_DIR)
```
Comment (creydr) [inline Makefile:119]: What are the reasons this target go removed?

File: Makefile [hunk 2]
```diff
@@ -115,8 +115,8 @@ whitespace-check:
...
+lint: $(GO)
```
Comment (rhrazdil) [inline Makefile:118]: Why not add lint as a new target?

**Review by rhrazdil — APPROVED** [review]
/approve
/lgtm
Thanks!

**Pull request merged**